src/model/types/index.ts

```diff
@@ -0,0 +1,49 @@
+import { Proof } from '../Proof.js'
```

Comment:
nice! that's better here.

Before we merge, we could add types for the payloads in here

Reply:
I added a CheckSpendablePayload type I don't think there is much sense to add types for the other payloads.